### PR TITLE
Feature members

### PR DESCRIPTION
**Pull request by ahmadchata**
## Rails-Members-Only

In this pull request, I did the following:

- Create an app where you can post with a title and a body.
- Will only be able to post if you are logged in.
- Non-members can see the posts but can't see their authors.

### CODE REVIEW

**Comment by lekegitrepo** [comment]
**_Hi @ahmadchata, I am marking this as invalid because of the following errors when I tried to install the dependencies :point_down:_** 

**`Gem::Ext::BuildError: ERROR: Failed to build gem native extension.`**
**`An error occurred while installing nokogiri (1.10.10), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.10.10' --source 'https://rubygems.org/'` succeeds before
bundling.`**

**Comment by ahmadchata** [comment]
-  Update README file for instructions if you encounter an error installing dependencies

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -10,6 +10,38 @@
...
+        <% else %>
+        <div class="btn-group">
+        <button class="btn btn-success">
+          <%= link_to "Sign In", new_user_session_path %>
+        </button>
+        <button class="btn btn-success">
+          <%= link_to "Sign Up", new_user_registration_path %>
+        </button>
+        </div>
+        <% end %>
```
Comment (lekegitrepo) [inline app/views/layouts/application.html.erb:42]: - **_Remove logic from the view_**

File: app/views/posts/_post.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      <strong><%= post.title %></strong>
+    </div>
+    <div class="card-body">
+    <blockquote class="blockquote mb-0">
+      <p><%= post.body %></p>
+      <% if user_signed_in? %>
+      <footer class="blockquote-footer">
+        <small><%= post.user.name %></small>
+      </footer>
+      <% end %>
```
Comment (lekegitrepo) [inline app/views/posts/_post.html.erb:15]: - **_Remove logic from the view_**

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class PostsController < ApplicationController
+  def index
+    @posts = Post.all.order('created_at DESC')
+  end
+
+  def new
+    @post = current_user.posts.build
+  end
```
Comment (lekegitrepo) [inline app/controllers/posts_controller.rb:8]: - **_The new action method should initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
#### Hi @ahmadchata :clap: :clap:
#### You have done a good job :+1: However, you need to consider the following changes(mentioned in my comments below) before this project can be approved.

**_Fix these and submit them for another review_**

#### _Reviewed by Temitayo Adeleke_

**Comment by ahmadchata** [comment]
## Changes made

- [x] Modified the post controller
- [x] Hid logic from views

**Review by kevincappuccino — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

## I really like your project! Congrats you did an excellent job, you have my approval to go ahead and merge this pull request!

_**Clean and modular code, following the best practices, keep it up!**_ 🥇


### As always, I wish you the very best and see you at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Pull request merged**